### DISCUSSION TITLE
Automerge with `forkMode: true` does not seem to work due to pending CI checks

### DISCUSSION THREAD

**Discussion by devversion** [renovatebot/renovate#21708] [Suggest an Idea]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

v32.99.8

### Please select which platform you are using if self-hosting.

github.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitHub

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Hey, we recently switched from the Mend-hosted Github app to self-hosted and enabled `forkMode: true`. Other than that, no other changes have been made. We realized that auto-merging stopped working.

In the logs, it seems like Renovate checks for CI statuses of pull requests before merging (which is a good thing!). It just seems to always log that there are zero status checks for a PR, even though there being various checks in the UI.

It's not clear from the documentation, but potentially automerge does not work with `forkMode` and this is intentional? My guess is that Renovate tries to determine the status from the upstream branch (which does not exist with fork mode), but instead should use the SHA from the PR.

Here is the PR associated with the logs from below: https://github.com/angular/ngcc-validation/pull/6830

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: PR is configured for automerge (repository=angular/ngcc-validation, branch=ng-renovate/main-swiper-8.x)
DEBUG: Checking schedule(at any time, America/Tijuana) (repository=angular/ngcc-validation, branch=ng-renovate/main-swiper-8.x)
DEBUG: No schedule defined (repository=angular/ngcc-validation, branch=ng-renovate/main-swiper-8.x)
DEBUG: resolveBranchStatus(branchName=ng-renovate/main-swiper-8.x, ignoreTests=false) (repository=angular/ngcc-validation, branch=ng-renovate/main-swiper-8.x)
DEBUG: getBranchStatus(ng-renovate/main-swiper-8.x) (repository=angular/ngcc-validation, branch=ng-renovate/main-swiper-8.x)
DEBUG: branch status check result (repository=angular/ngcc-validation, branch=ng-renovate/main-swiper-8.x)
       "state": "pending",
       "statuses": []
DEBUG: No check runs found (repository=angular/ngcc-validation, branch=ng-renovate/main-swiper-8.x)
       "result": {"total_count": 0, "check_runs": []}
DEBUG: Branch status yellow (repository=angular/ngcc-validation, branch=ng-renovate/main-swiper-8.x)
DEBUG: PR is not ready for merge (branch status is yellow) (repository=angular/ngcc-validation, branch=ng-renovate/main-swiper-8.x)
```

</details>

See above, for a small snippet.

https://github.com/angular/dev-infra/runs/7145316195?check_suite_focus=true

### Have you created a minimal reproduction repository?

No reproduction, but I have linked to a public repo where it occurs

**Comment by rarkins** [comment]
It wasn't really something looked at before because the assumption was that those using fork mode don't want to grant write permission to the upstream repo. 

Are you doing only PR based automerging? If so would the platformAutomerge feature satisfy you? That way GitHub does the actual merge once tests pass.

**Comment by devversion** [comment]
Thanks for the quick response. Yeah, agreed that this usage is definitely a little contradictory. i.e. having the argument of using forks, but then granting auto-merge permissions. 

To clarify why it's done this way: We just have a single self-hosted runner for most of our Angular repositories, set with `forkMode: true`, but then we also run this rather non-critical repository as part of it (where we have automerge enabled).

Thanks for the suggestion. I will try that and post back here. Feel free to close this issue as I understand that this might be out-of-scope for Renovate (it's good having this documented here though). Edit: Works as expected. thx
